portalnet/src/utp_controller.rs

```diff
@@ -15,33 +25,253 @@ use crate::discovery::UtpEnr;
 pub struct UtpController {
     pub inbound_utp_transfer_semaphore: Arc<Semaphore>,
     pub outbound_utp_transfer_semaphore: Arc<Semaphore>,
-    /// uTP socket.
     utp_socket: Arc<UtpSocket<UtpEnr>>,
+    metrics: OverlayMetricsReporter,
+}
+
+lazy_static! {
+    /// The default configuration to use for uTP connections.
+    pub static ref UTP_CONN_CFG: ConnectionConfig = ConnectionConfig { max_packet_size: 1024, ..Default::default()};
 }
 
 impl UtpController {
-    pub fn new(utp_transfer_limit: usize, utp_socket: Arc<UtpSocket<UtpEnr>>) -> Self {
+    pub fn new(
+        utp_transfer_limit: usize,
+        utp_socket: Arc<UtpSocket<UtpEnr>>,
+        metrics: OverlayMetricsReporter,
+    ) -> Self {
         Self {
             utp_socket,
             inbound_utp_transfer_semaphore: Arc::new(Semaphore::new(utp_transfer_limit)),
             outbound_utp_transfer_semaphore: Arc::new(Semaphore::new(utp_transfer_limit)),
+            metrics,
+        }
+    }
+
+    /// Connect with a peer given the connection id, and return the data received from the peer.
+    pub async fn connect_inbound_stream(
+        &self,
+        connection_id: u16,
+        peer: Enr,
+        trace: Option<QueryTrace>,
```

Comment:
nit: Passing the trace into this function is a bit off putting to me because it's not immediately clear why connecting an inbound stream would require a trace. After looking, I see that it's to pass it back out in an error in case the utp fails, but I wonder if we want to instead just add the trace to the error received in the overlay service so that this function's signature remains as simple as possible. 

That is....unless we want to add utp info to the trace. 🤔